vaxrank/epitope_prediction.py

```diff
@@ -77,24 +84,52 @@ def logistic_score(
 
         return logistic / normalizer
 
-def predict_epitopes(mhc_predictor, protein_fragment):
+# TODO(julia): move this code to probably topiary
+def compute_self_peptides(genome, ks):
+    logger.info("Computing self peptides...")
+    logger.info(genome)
+    # check if this exists
+    kmer_dict = defaultdict(set)
+    for k in ks:
+        for t in genome.transcripts():
+            if t.is_protein_coding:
+                p = t.protein_sequence
+                if p:
+                    for i in range(len(p) - k):
```

Comment:
If `p` is 9 amino acids and we want to generate reference 9mers then this expression would evaluate to `range(9 - 9)` = `range(0)`. Should this be `len(p) - k + 1`? I think this is my mistake from when we were prototyping earlier.